### ISSUE TITLE
How to debug this?

### ISSUE THREAD

**Issue by evandrocoan** [PyO3/maturin#269]
Deep down, on the development branch, this project https://github.com/ankitects/anki started using maturin for its stuff.

But when trying to use it here, I cannot figure out why `maturin` is not working:
```
maturin develop --release --strip
💥 maturin failed
  Caused by: Cargo metadata failed
  Caused by: No such file or directory (os error 2)
Makefile:17: recipe for target 'develop' failed
```
I tried to find for some verbose flag or anything else which could tell more about these errors as:
1. Which file was not found?
1. What cargo metadata failed?

**Comment by seanjensengrey** [comment]
Your subshell probably doesn't have `cargo` in the path.

**Comment by konstin** [comment]
> I tried to find for some verbose flag or anything else which could tell more about these errors as:
> * Which file was not found?
> 
> * What cargo metadata failed?

The command `cargo metadata` failed. Unfortunately, the "No such file or directory (os error 2)" comes directly from the standard library, which does not give us any more information. As @seanjensengrey said, `cargo` is probably missing, so I've changed the error message to `Cargo metadata failed. Do you have cargo in your PATH?`.

**Comment by evandrocoan** [comment]
Thanks for looking into it and improved the error message. I was using `mintty.exe` terminal and my Cygwin installation was bugged and even with cargo in the path, it was not being found. It was only finding cargo if I called it as  `cargo.exe` instead of just `cargo`. I was fixed after I ran again the Cygwin installer to rebase the Cygwin installation. But at the time a quicker fix was just use the `cmd.exe` terminal instead of the `mintty.exe` one.
